numpy/core/tests/test_multiarray.py

```diff
@@ -7943,7 +7943,8 @@ def __array__(self):
             assert_array_equal(res, base_arr)
             assert res is base_arr  # numpy trusts the ArrayLike
 
-        assert np.array(arr, copy=np._CopyMode.NEVER) is base_arr
+        with pytest.raises(ValueError):
+            np.array(arr, copy=np._CopyMode.NEVER)
```

Comment:
The fact that only this test change is needed gives me confidence that we did not break anything internally.